pkg/deploy/generator/resources.go

```diff
@@ -1573,3 +1579,197 @@ func (g *generator) rpVersionStorageAccount() *arm.Resource {
 	}
 }
+
+func (g *generator) devCIPool() *arm.Resource {
+	parts := []string{
+		fmt.Sprintf("base64ToString('%s')", base64.StdEncoding.EncodeToString([]byte("set -e\n\n"))),
+	}
+
+	for _, variable := range []string{
+		"azpToken",
+		"poolName",
+		"agentPrefix"} {
+		parts = append(parts,
+			fmt.Sprintf("'%s='''", strings.ToUpper(variable)),
+			fmt.Sprintf("parameters('%s')", variable),
+			"'''\n'",
+		)
+	}
+
+	trailer := base64.StdEncoding.EncodeToString([]byte(`yum -y update -x WALinuxAgent
+# install az cli
+sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+cat > /etc/yum.repos.d/azure-cli.repo <<'EOF'
+[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc
+EOF
+
+# install common tooling
+yum -y install rhui-azure-rhel7-devtools.noarch rhui-azure-rhel7.noarch rhui-azure-rhel7-eus.noarch
+yum -y install docker azure-cli jq rh-git29 rh-python36 gcc gpgme-devel libassuan-devel
+
+cat > /etc/profile.d/enablerh-git29.sh <<'EOF'
+#!/bin/bash
+if [ -x /usr/bin/scl_source ]; then
+  if [ -r /etc/scl/prefixes/rh-git29 ]; then
+	source scl_source enable rh-git29
+  fi
+fi
+EOF
+
+cat > /etc/profile.d/enablerh-python36.sh <<'EOF'
+#!/bin/bash
+if [ -x /usr/bin/scl_source ]; then
+  if [ -r /etc/scl/prefixes/rh-python36 ]; then
+  source scl_source enable rh-python36
+  fi
+fi
+EOF
+
+# install golang
+GO_VERSION=1.13.15
+curl -sL https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz -o /tmp/go${GO_VERSION}.linux-amd64.tar.gz
+mkdir -p /usr/local/go
+tar -C /usr/local/go -xzf /tmp/go${GO_VERSION}.linux-amd64.tar.gz --strip-components=1 go
+ln -s /usr/local/go/bin/* /usr/bin/
+echo "GOROOT=/usr/local/go" | tee -a /etc/environment
+
+# Install azp agent
+VSTS_AGENT=2.172.2
+wget https://vstsagentpackage.azureedge.net/agent/${VSTS_AGENT}/vsts-agent-linux-x64-${VSTS_AGENT}.tar.gz -O /tmp/vsts-agent-linux-x64-${VSTS_AGENT}.tar.gz
+mkdir /home/cloud-user/agent && cd /home/cloud-user/agent
+tar zxvf /tmp/vsts-agent-linux-x64-${VSTS_AGENT}.tar.gz
+./bin/installdependencies.sh
+sudo chown cloud-user:root -R /home/cloud-user/agent
+
+# configure agent
+AGENTNAME=$AGENTPREFIX-$HOSTNAME
+sudo -u cloud-user ./config.sh --unattended --url https://dev.azure.com/msazure --auth pat --token "$AZPTOKEN" --pool "$POOLNAME" --agent "$AGENTNAME"
+./svc.sh install
+
+# azure scripts do not work well with rhel7
+# TODO: Fix upstream in vsts repos
+cat > /home/cloud-user/agent/.path <<'EOF'
+/opt/rh/rh-python36/root/usr/bin:/opt/rh/rh-git29/root/usr/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/cloud-user/.local/bin:/home/cloud-user/bin
+EOF
+
+cat > /home/cloud-user/agent/.env <<'EOF'
+LANG=en_US.UTF-8
+LD_LIBRARY_PATH=/opt/rh/rh-python36/root/usr/lib64:/opt/rh/httpd24/root/usr/lib64
+PERL5LIB=/opt/rh/rh-git29/root/usr/share/perl5/vendor_perl
+HOME=/home/cloud-user
+EOF
```

Comment:
interested to know why these are needed

Reply:
Because MSFT agent runs in the custom context/shell interpreter and does not honour hosts configs for these values. It works on Ubuntu, but they are not friends with rhel :/ 
We ship python36 and git2 (hard requirement for VSTS) only ins RHSCL. So we need to make sure agent knows where to look for them when executing tests in msft shell. On rhel8 it is easier as these are in standard PATH and agent understands it. 

Painful.

Very.